src/introspective.jl

```diff
@@ -3,41 +3,45 @@ CACHETIMEOUT = 30 # s
 
 MAX_RETURN_SIZE = 20 # how many results to return at most
 
-
-function searchdocs(needle::String; loaded = true, mod = "Main",
-                    maxreturns = MAX_RETURN_SIZE, exportedonly = false,
-                    name_only = false)
+function searchdocs(needle::AbstractString; loaded::Bool = true, mod::Module = Main,
```

Comment:
So this PR changed the exported API a bit -- `mod` is now should be `Module` object instead of the previous implicit `String` specification.

Because of this, I would like to bump the version to `0.4.0` and then bump Atom.jl's compat of DocSeeker.jl accordingly after the new version released.